cpuid.go

```diff
@@ -554,6 +559,54 @@ func maxFunctionID() uint32 {
 	return a
 }
 
+func isHypervisorActive() bool {
+	_, _, i, _ := cpuid(0x1)
+	return i&(1<<31) != 0
+}
+
+func getHypervisorCpuid(ax uint32) string {
+	a, b, c, d := cpuid(ax)
+	var info [4]uint32 = [4]uint32{a, b, c, d}
+	name := strings.TrimRight(string((*[12]byte)(unsafe.Pointer(&info[1]))[:]), "\000")
+	return name
+}
+
+// see https://people.redhat.com/~rjones/virt-what/ for how full coverage is done
+// Vbox etc will need (optional - only do if root) dmi access
+func hypervisorName() string {
+	if !isHypervisorActive() {
+		return ""
+	}
+	// KVM has been caught to move its real signature to this leaf, and put something completely different in the
+	// standard location. So this leaf must be checked first.
+	// Sven removed it - in one test system, this leaf returns garbage :(
+	//if hv := getHypervisorCpuid(0x40000100); hv != "" {
+	//	return hv
+	//}
+
+	if hv := getHypervisorCpuid(0x40000000); hv != "" {
```

Comment:
Just `return getHypervisorCpuid(0x40000000)` - the remaining code does nothing.

Not sure if you should check availability of it somehow.